ImagePicker/CaptureSession.swift

```diff
@@ -793,25 +750,21 @@ private extension CaptureSession {
     func processStopVideoCapturing(delegate: VideoCaptureDelegate, outputURL: URL) {
         removeReferenceToVideoCaptureDelegate()
         DispatchQueue.main.async { [weak self] in
-            guard let `self` = self else { return }
             if delegate.isBeingCancelled {
-                self.videoRecordingDelegate?.captureSessionDidCancelVideoRecording(self)
-            }
-            else {
-                self.videoRecordingDelegate?.captureSessionDid(self, didFinishVideoRecording: outputURL)
+                self?.videoRecordingDelegate?.captureSessionDidCancelVideoRecording(self!)
+            } else {
+                self?.videoRecordingDelegate?.captureSessionDid(self!, didFinishVideoRecording: outputURL)
             }
         }
     }
 
     func processFailVideoCapturing(delegate: VideoCaptureDelegate, outputURL: URL, error: Error) {
         removeReferenceToVideoCaptureDelegate()
         DispatchQueue.main.async { [weak self] in
```

Comment:
why to use weak self while dispatching ? are you sure the retain cycle is created ?